### DISCUSSION TITLE
Unexpected beatmap selection when changing the sorting and ruleset

### DISCUSSION THREAD

**Discussion by felipemarins** [ppy/osu#27108] [Q&A]
It seems like song select doesn't account for the sorting selected after changing it while staying on the same beatmap and selects a ~random~ beatmap based on the position of the previous sorting when changing rulesets.

Steps to reproduce:
- Select a beatmap with **sorting A**
- Change to **sorting B**
- Change ruleset
- The selected beatmap does not follow the position of the other beatmap in **sorting B**

I tested it with sorting by Artist, Author, Date Added, Date Submitted and Title. It seems to only follow the beatmap position on sorting B when changing the sorting before selecting the beatmap.

**Comment by bdach** [comment]
I'm not sure what this report is trying to say. Looks to be sorting correctly to me.

Please upload video of this explaining what you're seeing. Or use a concrete example rather than "sorting A" and "sorting B".

**Reply by felipemarins** [reply]
Sorry for the confusion. It's not the sorting that is incorrect, but the selected beatmap after the ruleset change. As a video would be very laggy on my laptop, I'll try the second option but with images.

### Expected behaviour

1. Change to sorting by **artist** 
![image](https://github.com/ppy/osu/assets/53406701/eb2a3ea7-e01a-46b4-939f-38ba445eb3e3)
2. Select another beatmap 
![image](https://github.com/ppy/osu/assets/53406701/d1eb4b31-a965-46f7-9efa-08d9321e66ad)
3. Change the ruleset
4. The selected beatmap is on the same position 
![image](https://github.com/ppy/osu/assets/53406701/a8220e43-b049-467a-ad92-bb6323f37d73)

### What I was trying to report

1. Select a beatmap
![image](https://github.com/ppy/osu/assets/53406701/2dc91f36-0296-46b6-853d-86fda6b69b5e)
2. Change to sorting by **artist**
![image](https://github.com/ppy/osu/assets/53406701/c7ec645c-2da0-4bf4-80b9-4766168143ac)
3. Change the ruleset
4. The selected beatmap is not on the same position
![image](https://github.com/ppy/osu/assets/53406701/822a63fd-4f04-46a7-aaa9-354339435eee)

It seems that the position of the beatmap selected when changing rulesets is roughly the same position that the previous beatmap was on previous sorting (on sorting by date added).

**Reply by felipemarins** [reply]
I finally got a video:

https://github.com/ppy/osu/assets/53406701/e7251c6d-2dc0-434c-b3bf-343342c6043c
